unstructured/ingest/connector/azure_cognitive_search.py

```diff
@@ -0,0 +1,118 @@
+        )
+
+    def conform_dict(self, data: dict) -> None:
+        from dateutil import parser  # type: ignore
+
+        data["id"] = str(uuid.uuid4())
+
+        if points := data.get("metadata", {}).get("coordinates", {}).get("points"):
+            data["metadata"]["coordinates"]["points"] = json.dumps(points)
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+        if record_locator := data.get("metadata", {}).get("data_source", {}).get("record_locator"):
+            data["metadata"]["data_source"]["record_locator"] = json.dumps(record_locator)
+        if last_modified := data.get("metadata", {}).get("last_modified"):
+            data["metadata"]["last_modified"] = parser.parse(last_modified).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+        if date_created := data.get("metadata", {}).get("data_source", {}).get("date_created"):
+            data["metadata"]["data_source"]["date_created"] = parser.parse(date_created).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+        if date_modified := data.get("metadata", {}).get("data_source", {}).get("date_modified"):
+            data["metadata"]["data_source"]["date_modified"] = parser.parse(date_modified).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+        if date_processed := data.get("metadata", {}).get("data_source", {}).get("date_processed"):
+            data["metadata"]["data_source"]["date_processed"] = parser.parse(
+                date_processed,
+            ).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        if regex_metadata := data.get("metadata", {}).get("regex_metadata"):
+            data["metadata"]["regex_metadata"] = json.dumps(regex_metadata)
+        if page_number := data.get("metadata", {}).get("page_number"):
+            data["metadata"]["page_number"] = str(page_number)
+
+    def write(self, docs: t.List[BaseIngestDoc]) -> None:
+        json_list = []
+        for doc in docs:
+            local_path = doc._output_filename
+            with open(local_path) as json_file:
+                # TODO element id not a sufficient unique id to use
+                json_content = json.load(json_file)
+                for content in json_content:
+                    self.conform_dict(data=content)
+                logger.info(
+                    f"appending {len(json_content)} json elements from content in {local_path}",
+                )
+                json_list.extend(json_content)
+        logger.info(
+            f"writing {len(json_list)} documents to destination "
+            f"index at {self.write_config.index}",
+        )
+        try:
+            results = self.client.upload_documents(documents=json_list)
+
+        except azure.core.exceptions.HttpResponseError as http_error:
+            raise WriteError(f"http error: {http_error}") from http_error
```

Comment:
do we know the difference between connectivity and unrecoverable issue writing? ...i.e. should a caller wrap WriteError in retry? mostly with custom errors want to distinguish based on potentially actionable issues. if it's connectivity, retrying might help. if it's an issue with writing because something is invalid, retrying won't help

Reply:
Can incorporate with the new retry strategy introduced in the other PR